From 2cde03fb8ff70fc2f6910a2240e65fe74b5789a3 Mon Sep 17 00:00:00 2001
From: Cristinel Panfir <cristinel.panfir@nxp.com>
Date: Wed, 8 Sep 2021 13:57:32 +0300
Subject: [PATCH] vx-delegate/CMakeLists.txt: Fix TIM_VX library path

Use CMAKE variable to avoid hardcoded lib path.

Multilib env error:

Log data follows:
| DEBUG: Executing shell function do_compile
| NOTE: VERBOSE=1 cmake --build /opt/work/multilib/5.10.52-2.1.0/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/tensorflow-lite/2.5.0-r0/build --target all --
| ninja: error: '/opt/work/multilib/5.10.52-2.1.0/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/tensorflow-lite/2.5.0-r0/recipe-sysroot/usr/lib/libtim-vx.so', needed by 'libtensorflow-lite.so.2.5.0', missing and no known rule to make it
| WARNING: exit code 1 from a shell command.
| ERROR: Execution of '/opt/work/multilib/5.10.52-2.1.0/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/tensorflow-lite/2.5.0-r0/temp/run.do_compile.2662673' failed with exit code 1:
| ninja: error: '/opt/work/multilib/5.10.52-2.1.0/build/tmp/work/cortexa72-cortexa53-crypto-poky-linux/tensorflow-lite/2.5.0-r0/recipe-sysroot/usr/lib/libtim-vx.so', needed by 'libtensorflow-lite.so.2.5.0', missing and no known rule to make it
| WARNING: exit code 1 from a shell command.
|

Signed-off-by: Cristinel Panfir <cristinel.panfir@nxp.com>
---
 tensorflow/lite/delegates/vx-delegate/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tensorflow/lite/delegates/vx-delegate/CMakeLists.txt b/tensorflow/lite/delegates/vx-delegate/CMakeLists.txt
index b2a67345a71..66c1af6caab 100644
--- a/tensorflow/lite/delegates/vx-delegate/CMakeLists.txt
+++ b/tensorflow/lite/delegates/vx-delegate/CMakeLists.txt
@@ -46,7 +46,7 @@ endif()
 
 message("=== Building with TIM_VX_LIBRIRIES from ${TIM_VX_INSTALL} ===")
 include_directories(${TIM_VX_INSTALL}/include)
-list(APPEND TFLITE_DELEGATES_VX_DEPENDENCIES ${TIM_VX_INSTALL}/lib/libtim-vx.so)
+list(APPEND TFLITE_DELEGATES_VX_DEPENDENCIES ${TIM_VX_INSTALL}/${CMAKE_INSTALL_LIBDIR}/libtim-vx.so)
 
 message("=== Linking with EXTERNAL_VIV_SDK from ${EXTERNAL_VIV_SDK} ===")
 if( EXISTS ${EXTERNAL_VIV_SDK}/drivers)
-- 
2.25.1

